### PR TITLE
tpm: Fix eventlog replay for Intel TXT machines

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/tpm/fu-tpm-eventlog-common.c [hunk 1]
```diff
@@ -175,6 +175,22 @@ fu_tpm_eventlog_calc_checksums(GPtrArray *items, guint8 pcr, GError **error)
 		FuTpmEventlogItem *item = g_ptr_array_index(items, i);
 		if (item->pcr != pcr)
 			continue;
+
+		/* if TXT is enabled then the first event for PCR0 should be a StartupLocality */
+		if (item->kind == EV_NO_ACTION && item->pcr == 0 && item->blob != NULL && i == 0) {
+			gsize bufsz;
+			const guint8 *buf = g_bytes_get_data(item->blob, &bufsz);
+			if (bufsz == 17 && buf[bufsz - 2] == 0) {
```
Comment (superm1) [inline plugins/tpm/fu-tpm-eventlog-common.c:183]: still want to decode the magic buffer size here so we better understand them when the size changes.

Comment (hughsie) [reply plugins/tpm/fu-tpm-eventlog-common.c:183]: @flanfly is this encoded into the event data or hardcoded? Thanks.

Comment (hughsie) [reply plugins/tpm/fu-tpm-eventlog-common.c]: @superm1 how about that? It's pretty self documenting now...

**Review by superm1 — APPROVED** [review]

File: plugins/tpm/fu-tpm-eventlog-common.c (hunk 1 above)
Comment (flanfly) [reply plugins/tpm/fu-tpm-eventlog-common.c]: The event data is always "StartupLocality\x00\xXX" whereas \xXX is the TPM 2.0 locality the Startup command was issued from (should be either 0 or 3).

**Pull request merged**